### ISSUE TITLE
[BUG] TimeSeries error when loading with a different OS.

### ISSUE THREAD

**Issue by pedronarloch** [autogluon/autogluon#3318]
**Bug Report Checklist**

- [ ] I provided code that demonstrates a minimal reproducible example.  
- [X] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [X] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

In https://github.com/autogluon/autogluon/pull/2865 there is a fix for TABULAR where incompatibilities exist when saving/loading models. The fix works for TABULAR but is not working for TIME SERIES because it was only implemented in `core/src/autogluon/core/trainer/abstract_trainer.py`. When inspecting `timeseries/src/autogluon/abstract_trainer.py`, the non-working code is still there. Please compare the create_contexts(...) method for both files.

**Expected behavior**

When loading time series models, it runs the same fix from Tabular.

**To Reproduce**

```python
# core/src/autogluon/core/trainer/abstract_trainer.py
def create_contexts(self, path_context: str) -> (str, dict):
    self.path = PathConverter.to_current(self.path)
    path = path_context
    model_paths = self.get_models_attribute_dict(attribute='path')
    model_paths = {model: PathConverter.to_current(model_path) for model, model_path in model_paths.items()}
    for model, prev_path in model_paths.items():
        prev_path = os.path.abspath(prev_path) + os.path.sep
        abs_path = os.path.abspath(self.path) + os.path.sep
        model_local_path = prev_path.split(abs_path, 1)[1]
        new_path = path + model_local_path
        model_paths[model] = new_path

    return path, model_paths
```

Vs.

```python
# timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py
def create_contexts(self, path_context: str) -> Tuple[str, dict]:
    path = path_context
    # TODO: consider keeping track of model path suffixes in model_graph instead
    # TODO: of full paths
    model_paths = self.get_models_attribute_dict(attribute="path")
    for model, prev_path in model_paths.items():
        model_local_path = prev_path.split(self.path, 1)[1]
        new_path = path + model_local_path
        model_paths[model] = new_path

    return path, model_paths
```

**Screenshots / Logs**

`IndexError: list index out of range`

python3.9/site-packages/autogluon/timeseries/trainer/abstract_trainer.py:126 in create_contexts
│    123 │   │   # TODO: of full paths                                                                         │
│    124 │   │   model_paths = self.get_models_attribute_dict(attribute="path" │
│    125 │   │   for model, prev_path in model_paths.items():                                │
│ ❱  126 │   │   │   model_local_path = prev_path.split(self.path, 1)[1]                   │
│    127 │   │   │   new_path = path + model_local_path                                          │
│    128 │   │   │   model_paths[model] = new_path                                                 │

**Installed Versions**

`poetry add autogluon --allow-prereleases`

```python
autogluon              : 0.7.1b20230614
autogluon.common       : 0.7.1b20230614
autogluon.core         : 0.7.1b20230614
autogluon.features     : 0.7.1b20230614
autogluon.multimodal   : 0.7.1b20230614
autogluon.tabular      : 0.7.1b20230614
autogluon.timeseries   : 0.7.1b20230614
```

</details>

**Comment by shchur** [comment]
Thanks for reporting this problem! We won't be able to fix it for the v0.8 release due to the code freeze, but will try to address it later.

**Comment by shchur** [comment]
Should be fixed by #3260.
